packages/docusaurus-theme-classic/src/theme/Details/index.tsx

```diff
@@ -9,6 +9,8 @@ import React from 'react';
 import clsx from 'clsx';
 import {Details as DetailsGeneric} from '@docusaurus/theme-common';
 import type {Props} from '@theme/Details';
+// Ensure that the default details style is properly overridden
+import '@docusaurus/theme-common/lib/components/Details/styles.module.css';
```

Comment:
We unfortunately have to do this, to make theme-common tree-shakeable at all. Maybe we should have different `exports` fields for all side-effect-ful exports (like `@docusaurus/theme-common/Details`), or simply declare this as the official way to acquire default styles for `Details`?

Reply:
🤷‍♂️ if we do this maybe we should remove the css module import in theme common?

Note `useKeyboardNavigation` also imports css

Reply:
Aye, but that one's not impacted by CSS insertion order, so as long as it is loaded (which, well, means `useKeyboardNavigation.ts` is used), it works properly.

My idea is that we should not export `Details` from `src/index.tsx`, but create a separate `exports` entry point instead.

Reply:
What I understand is that the CSS order problem is related to theme/Details vs theme-common/Details right?

This kind of problem is going to be solved soon once CSS cascade layers support improves (already quite good https://caniuse.com/css-cascade-layers)

```css
@layer infima, theme-common, theme, site;
```

In the meantime maybe we could just add `!important` to the 4 rules that override theme-common?

Reply:
Mmm, you can see I obviously don't like `!important` because that makes site customizations harder, if not impossible...

Reply:
yes agree, your solution is probably a better temporary workaround